src/Microsoft.OpenApi.OData.Reader/Common/Utils.cs

```diff
@@ -270,5 +271,131 @@ internal static void AddCustomAttributesToExtensions(this IDictionary<string, IO
 
             return false;
         }
+
+
+        /// <summary>
+        /// Gets the entity type of the target <paramref name="segment"/>.
+        /// </summary>
+        /// <param name="segment">The target <see cref="ODataSegment"/></param>
+        /// <returns>The entity type of the target <paramref name="segment"/></returns>
+        internal static IEdmEntityType EntityTypeFromPathSegment(this ODataSegment segment)
```

Comment:
```suggestion
        private static IEdmEntityType EntityTypeFromPathSegment(this ODataSegment segment)
```

Reply:
This extension method will def. be reused outside this class.